C error when trying to initialize generic fixed array type

### Describe the bug

Trying to compile
```v
module main

fn init_type[T]() T {
	return T{}
}

fn main() {
	init_type[[3]int]()
}
```
gives
```
/tmp/v_1002/src.170089672244163027.tmp.c:12624: error: '{' expected (got ";")
```

### Reproduction Steps

Try to compile the above code

### Expected Behavior

Compile successfully

### Current Behavior

```
/tmp/v_1002/src.170089672244163027.tmp.c:12624: error: '{' expected (got ";")
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.3 4628fe7

### Environment details (OS name and version, etc.)

```yaml
V full version: V 0.4.3 4bc9a8f.4628fe7
OS: linux, Linux version 6.5.11-300.fc39.x86_64 (mockbuild@d23353abed4340e492bce6e111e27898) (gcc (GCC) 13.2.1 20231011 (Red Hat 13.2.1-4), GNU ld version 2.40-13.fc39) #1 SMP PREEMPT_DYNAMIC Wed Nov  8 22:37:57 UTC 2023
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-4310U CPU @ 2.00GHz

getwd: /home/user/Projects/v-testing/embed_dir
vexe: /home/user/.local/lib/v/v
vexe mtime: 2023-11-14 10:16:10

vroot: OK, value: /home/user/.local/lib/v
VMODULES: OK, value: /home/user/.vmodules
VTMP: OK, value: /tmp/v_1002

Git version: git version 2.41.0
Git vroot status: weekly.2023.45.1-25-g4628fe76
.git/config present: true

CC version: cc (GCC) 13.2.1 20231011 (Red Hat 13.2.1-4)
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/return.v
+++ b/vlib/v/checker/return.v
@@ -34,6 +34,10 @@ fn (mut c Checker) return_stmt(mut node ast.Return) {
 		}
 	}
 	expected_type_sym := c.table.sym(expected_type)
+	if expected_type_sym.info is ast.ArrayFixed {
+		c.table.find_or_register_array_fixed(expected_type_sym.info.elem_type, expected_type_sym.info.size,
+			expected_type_sym.info.size_expr, true)
+	}
 	if node.exprs.len > 0 && c.table.cur_fn.return_type == ast.void_type {
 		c.error('unexpected argument, current function does not return anything', node.exprs[0].pos())
 		return

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4896,7 +4896,7 @@ fn (mut g Gen) return_stmt(node ast.Return) {
 	}
 
 	// got to do a correct check for multireturn
-	sym := g.table.sym(g.fn_decl.return_type)
+	sym := g.table.sym(g.unwrap_generic(g.fn_decl.return_type))
 	mut fn_ret_type := g.fn_decl.return_type
 	if sym.kind == .alias {
 		unaliased_type := g.table.unaliased_type(fn_ret_type)

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -227,7 +227,7 @@ fn (mut g Gen) gen_fn_decl(node &ast.FnDecl, skip bool) {
 	mut name := g.c_fn_name(node)
 	mut type_name := g.typ(g.unwrap_generic(node.return_type))
 
-	ret_sym := g.table.sym(node.return_type)
+	ret_sym := g.table.sym(g.unwrap_generic(node.return_type))
 	if node.return_type.has_flag(.generic) && ret_sym.kind == .array_fixed {
 		type_name = '_v_${type_name}'
 	} else if ret_sym.kind == .alias && !node.return_type.has_flag(.option) {

--- a/vlib/v/tests/generics_return_fixed_array_test.v
+++ b/vlib/v/tests/generics_return_fixed_array_test.v
@@ -1,0 +1,9 @@
+fn init_type[T]() T {
+	return T{}
+}
+
+fn test_generics_return_fixed_array() {
+	r1 := init_type[[3]int]()
+	println(r1)
+	assert '${r1}' == '[0, 0, 0]'
+}

